burnman/planet.py

```diff
@@ -7,206 +7,834 @@
+                    oneprop.append(layer.evaluate([prop],r))
+                if all is None:
+                    all = np.array(oneprop)
+                else:
+                    all = np.vstack((all,np.array(oneprop)))
+        return all
+
+    def set_pressure_mode( self, pressure_mode='self-consistent', pressures=None, pressure_top=0.,
+              gravity_bottom=0., n_max_iterations=50, max_delta = 1.e-5):
+        """
+        Sets the pressure mode of the planet by user-defined values are in a self-consistent fashion.
+        pressure_mode is 'user-defined' or 'self-consistent'.
+        The default for the planet is self-consistent, with zero pressure at the surface and zero pressure at the center.
+        
+        Parameters
+        ----------
+        pressure_mode : string
+        This can be set to 'user-defined' or 'self-consistent'
+        pressures : array of floats
+        Pressures (Pa) to set layer to ('user-defined'). This should be the same length as defined radius array for the layer
+        pressure_top : float
+        Pressure (Pa) at the top of the layer.
+        gravity_bottom : float
+        gravity (m/s^2) at the bottom the layer
+        n_max_iterations : int
+        Maximum number of iterations to reach self-consistent pressures (default = 50)
+        """
+        self.reset()
+        assert(pressure_mode == 'user-defined' or pressure_mode == 'self-consistent')
+        
+        
+        self.pressure_mode = pressure_mode
+        self.gravity_bottom = gravity_bottom
+ 
+        if pressure_mode == 'user-defined':
+            assert(len(pressures) == len(self.radii))
+            self._pressures = pressures
+            warnings.warn(
+                "By setting the pressures in Planet it is unlikely to be self-consistent")
+
+        if pressure_mode == 'self-consistent':
+            self.pressure_top = pressure_top
+            self.n_max_iterations = n_max_iterations
+            self.max_delta = max_delta
+                
+                
+    def make(self):
+        """
+        This routine needs to be called before evaluating any properties. If pressures and temperatures are self-consistent, they
+        are computed across the planet here. Also initializes an array of materials in each Layer to compute properties from.
+        """
+        
+        self.reset()
         for layer in self.layers:
-            mypressures = self.pressures[layer.n_start: layer.n_end]
-            mytemperatures = self.temperatures[layer.n_start: layer.n_end]
+            assert(layer.temperature_mode is not None)
+
+
+        if self.pressure_mode == 'user-defined':
+            self._temperatures = self._evaluate_temperature(self._pressures)
+
+        if self.pressure_mode == 'self-consistent':
+
+            new_press = self.pressure_top + \
+                (-self.radii + max(self.radii)) * \
+                2.e5  # initial pressure curve guess
```

Comment:
2.e5 is a really big number. Is it possible to make a more reasonable guess? Otherwise lots of low pressure EoSes are going to fail here.